.github/workflows/run_all_frameworks.yml

```diff
@@ -0,0 +1,156 @@
+name: Run All Frameworks
+
+on: pull_request
+
+
+jobs:
+
+  detect_changes:  # Figure out which frameworks need to be evaluated based on which files are changed.
+    name: Detect Changes
+    runs-on: ubuntu-latest
+    outputs:
+      frameworks: ${{ steps.find-required-tests.outputs.frameworks }}
+      tasks: ${{ steps.find-required-tests.outputs.tasks }}
+      benchmark: ${{ steps.find-required-tests.outputs.benchmark }}
+      skip_baseline: ${{ steps.find-required-tests.outputs.skip_baseline }}
+      skip_evaluation: ${{ steps.find-required-tests.outputs.skip_evaluation }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: pull base branch
+        run: |
+          git fetch --all
+          git checkout -b $GITHUB_BASE_REF origin/$GITHUB_BASE_REF
+          git checkout $GITHUB_HEAD_REF
+      - id: find-required-tests
+        name: Detect Common Changes  # detect if any changes occurred that should trigger all frameworks.
+        run: |
+          changed_files=$(git diff --name-only HEAD..$GITHUB_BASE_REF)
+          common_files="amlb/** "\
+          "frameworks/shared/** "\
```

Comment:
what about including `config.yaml`? A change/renaming in this file can have an impact on the framework behaviour.
Or even include the whole `resources` folder to remain on the safe side.